hw17/html_render.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+
+"""
+Python class example.
+
+"""
+
+
+class Element(object):
+    IND_LEVEL = "    "
+    name = "html"
+
+    def __init__(self, content="", **kwargs):
+        self.attributes = kwargs
+        self.attr = ""
+        self.children = [content] if content else []
+        for attr_key, attr_val in self.attributes.items():
+            self.attr += "{key}=\"{val}\"".format(key=attr_key, val=attr_val)
+
+    def append(self, new_child):
+        self.children.append(new_child)
+
+    indent = "    "
+
+    def render(self, out_file, indent=""):
+        out_file.write(self.indent + "<" + self.name + self.attr + ">\n")
+
+        for child in self.children:
+            try:
+                child.render(out_file, self.indent + Element.IND_LEVEL)
+            except AttributeError:
+                out_file.write(self.indent + Element.IND_LEVEL + child + "\n")
+
+        out_file.write(self.indent + "</" + self.name + ">\n")
+
+
+class Html(Element):
+
+    name = "html"
+
+    def __init__(self, content=""):
+        Element.__init__(self, name="html")
+        self.indent = ""
+
+    def render(self, out_file, indent=""):
+        out_file.write("<!DOCTYPE html>\n")
+        Element.render(self, out_file)
+
+
+class Head(Element):
+
+    name = "head"
+
+
+class OneLineTag(Element):
+    def render(self, out_file, content):
+        out_file.write(self.indent + "<" + self.name + ">")
```

Comment:
You're missing the `self.attr` part of your HTML. That's why your <a> tag isn't working.